### PR TITLE
Reserve resources for calico typha

### PR DESCRIPTION
**Pull request by zonzamas**
Calico-typha does not have any reserved resource so it works in a best effort mode. Since it is a 'system service' it should have some resources reserved. What do you think?

### CODE REVIEW

**Comment by davidmccormick** [comment]
Good idea!  In which case they should be configurable via the cluster.yaml so that users can increase them if the defaults prove too small.  Can you add them in as options?

**Comment by DominicGunn** [comment]
Any chance you could sign the CLA and address David's comments @zonzamas?

**Comment by zonzamas** [comment]
Hi, Im have it sing already with the other email I used to use to contribute to kube-aws. I will definitely revisit this when I have some free time or when I need it for my current company.

**Comment by zonzamas** [comment]
I signed it

**Comment by dominicgunn** [comment]
Thanks @zonzamas, I'd still like to see this be configurable through `cluster.yaml` before merging. Could you take a look at doing that?

**Comment by zonzamas** [comment]
Sure, how about this?

File: builtin/files/userdata/cloud-config-controller [hunk 1]
```diff
@@ -1449,6 +1449,13 @@ write_files:
                 #  value: "true"
                 #- name: TYPHA_PROMETHEUSMETRICSPORT
                 #  value: "9093"
+              resources:
```
Comment (davidmccormick) [inline builtin/files/userdata/cloud-config-controller]: What you could do here is to put your default values into the function NewDefaultCluster() within pkg/api/cluster.go?  Then you don't need to do the conditional in the template... could you do that?

**Comment by davidmccormick** [comment]
Hi, many thanks - would you be so kind as to cherry-pick this change and PR to our v0.13.x branch and v0.14.x branch?  That way your change will go out with those upcoming updates rather than when v0.15.x is released.  Thanks for your contribution! 🙏

**Pull request merged**